src/library.js

```diff
@@ -2388,6 +2387,25 @@ mergeInto(LibraryManager.library, {
   _emscripten_get_now_is_monotonic__deps: ['$nowIsMonotonic'],
   _emscripten_get_now_is_monotonic: () => nowIsMonotonic,
 
+  __emscripten_atomics_sleep: (ms) => {
+    try {
+      Atomics.wait(waitBuffer, 0, 0, ms);
+      return 1;
+    } catch (_) {
+      return 0;
+    }
+  },
+  __emscripten_atomics_sleep__postset: `
+    var waitBuffer;
+    try {
+      var SharedArrayBuffer = new WebAssembly.Memory({"shared":true,"initial":0,"maximum":0}).buffer.constructor;
+      waitBuffer = new Int32Array(new SharedArrayBuffer(4));
+    } catch (_) { }
```

Comment:
Instead of creating a new SAB, it would be better to just sleep on the original wasm heap.

Reply:
Please add some comments in the source code here - this is definitely a confusing area...

Reply:
Why do we need to use `WebAssembly.memory` here?   Why not just use `SharedArrayBuffer` directly?

If we do need to go this route maybe its best not to shadow the global `SharedArrayBuffer`.

If there is any way to do this without a try/catch that would be great.